components/input/style/mixin.less

```diff
@@ -178,7 +181,7 @@
     &:hover {
       z-index: 1;
       border-right-width: 1px;
-      .@{ant-prefix}-input-search-with-button & {
```

Comment:
The resulting css styles are different when the `@ant-prefix` is overwritten in a `.less` file like here:

https://github.com/mll-lab/react-components/blob/master/src/antd.less#L4

So we want to use all the components like that:


antd.less:
```
@ant-prefix: 'my-prefix';
```

app.tsx:
```
...
<ConfigProvider prefixCls="my-prefix">
  ...
</ConfigProvider>
...
```

About the fix in this PR:

When `ant-prefix` is overwritten like described here in line 181, the custom prefix won't result in a css style like `my-prefix-search-with-button`. Instead, the `z-index: 0;` (see line 182) won't be applied on an input with the class `my-prefix-search-with-button`, because it doesn't exist.

I'm not good with LESS, but what I observed is that the following variants lead to a different result:

Variant 1 (aliased):
```
@input-prefix-cls: ~'@{ant-prefix}-input';
...
.@{input-prefix-cls}-search-with-button & {
```

Variant 2 (original):
```
.@{ant-prefix}-search-with-button & {
```

There are still quite many components left which require a similar fix: just need to search for `@{ant-prefix}`.

Maybe this issue can be solved much easier and not component-wise, but I didn't find a better solution. Do you know one?

Would it be possible for the future to also add some visual tests (screenshot diff) for the usage of `ant-prefix`?